lib/fog/vsphere/models/compute/virtual_machine_config_summary.rb

```diff
@@ -0,0 +1,35 @@
+module Fog
+  module Compute
+    class Vsphere
+      # https://www.vmware.com/support/developer/vc-sdk/visdk41pubs/ApiReference/vim.vm.Summary.ConfigSummary.html
+      class VirtualMachineConfigSummary < Fog::Model
+        attribute :annotation
+        attribute :cpuReservation
+        attribute :dynamicProperty
+        attribute :ftInfo #FaultToleranceConfigInfo
+        attribute :guestFullName
+        attribute :guestId
+        attribute :installBootRequired
+        attribute :instanceUuid
+        attribute :memoryReservation
+        attribute :memorySizeMB
+        attribute :name
+        attribute :numCpu
+        attribute :numEthernetCards
+        attribute :numVirtualDisks
+        attribute :product #VAppProductInfo
+        attribute :template
+        attribute :uuid
+        attribute :vmPathName
+
+        def initialize(attributes={})
+          super
+          #TODO: Other objects get initialized here
+          #TODO: For now they will be their rbvmomi counterparts
+          #TODO: self.ftInfo = FaultToleranceConfigInfo.new(attributes[:ftInfo].props) if attributes[:ftInfo] && attributes[:ftInfo].props
+          #TODO: self.product = VAppProductInfo.new(attributes[:product].props) if attributes[:product] && attributes[:product].props
+        end
```

Comment:
With the exception of the two objects I haven't defined yet, ftInfo and product, all of the other attributes are basic types (strings, bools, and ints).  I figured those was useful. If you want to wait until those are defined I can remove this.